components/UploadForm.vue

```diff
@@ -0,0 +1,671 @@
+                  name="i-heroicons-trash"
+                  class="cursor-pointer text-red-500"
+                  @click="handleDeleteFile(index)"
+                />
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    </div>
+    <div v-if="uploadStatus" class="w-full">
+      <div class="space-y-3">
+        <div class="flex justify-between items-center">
+          <UBadge color="Badge" variant="soft">
+            {{ uploadStatus }}
+          </UBadge>
+          <p class="text-xs text-gray-500">
+            請勿關閉此視窗，直到上傳完成。
+          </p>
+        </div>
+        <UProgress
+          animation="carousel"
+          color="primary"
+          class="w-full"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from 'pinia'
+import exifr from 'exifr'
+import ePub from 'epubjs'
+import { BigNumber } from 'bignumber.js'
+import { fileToArrayBuffer, digestFileSHA256, calculateIPFSHash } from '~/utils/index'
+import { useFileUpload } from '~/composables/useFileUpload'
+import {
+  estimateBundlrFilePrice,
+  uploadSingleFileToBundlr
+} from '~/utils/arweave'
+import { sendLIKE } from '~/utils/cosmos' // You'll need to implement this
+import { useWalletStore } from '~/stores/wallet'
+import { useBookStoreApiStore } from '~/stores/book-store-api'
+
+const UPLOAD_FILESIZE_MAX = 200 * 1024 * 1024
+const IMAGE_MIME_TYPES = ['image/jpeg', 'image/png', 'image/gif', 'image/webp']
+
+const store = useWalletStore()
+const { wallet, signer } = storeToRefs(store)
+const { initIfNecessary } = store
+const bookStoreApiStore = useBookStoreApiStore()
+const { token } = storeToRefs(bookStoreApiStore)
+
+const { getFileType } = useFileUpload()
+const fileRecords = ref([])
+const isSizeExceeded = ref(false)
+const isDragging = ref(false)
+const epubMetadataList = ref([])
+
+const arweaveFee = ref(new BigNumber(0))
+const uploadStatus = ref('')
+const arweaveFeeMap = ref({})
+const arweaveFeeTargetAddress = ref('')
+const sentArweaveTransactionInfo = ref(new Map())
+const isOpenWarningSnackbar = ref(false)
+const error = ref('')
+const balance = ref(new BigNumber(0))
+const signDialogError = ref('')
+const numberOfSignNeeded = ref(0)
+const signProgress = ref(0)
+
+const emit = defineEmits(['arweaveUploaded', 'submit'])
+
+const computedFormClasses = computed(() => [
+  'block',
+  'flex',
+  'w-full',
+  fileRecords.value.length ? 'max-w-[320px]' : '',
+  'flex-col',
+  'justify-center',
+  isSizeExceeded.value ? 'bg-transparent' : '',
+  'items-center',
+  'justify-between',
+  'p-[28px]',
+  'mb-[12px]',
+  'border-[2px]',
+  'border-dashed',
+  'border-gray-300',
+  'rounded-[12px]',
+  'text-gray-500',
+  'cursor-pointer',
+  'bg-gray-100',
+  'hover:bg-gray-200'
+])
+
+watch(fileRecords, async (newFileRecords) => {
+  if (newFileRecords.length) {
+    uploadStatus.value = 'loading'
+    await estimateArweaveFee()
+    uploadStatus.value = ''
```

Comment:
Should we add `try finally` here?

Reply:
[🥅 Handle errors during file upload estimation](https://github.com/likecoin/nft-book-press/pull/290/commits/70e265de6554274271489dc3bbe773f200d5015d)